internal/client/client.go

```diff
@@ -0,0 +1,121 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/deepmap/oapi-codegen/pkg/securityprovider"
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/pkg/errors"
+)
+
+func New(ctx context.Context, apiKey, apiEndpoint, version string, opts ...ClientOption) (*ClientWithResponses, error) {
```

Comment:
This looks much more like our catalog-importer client now: https://github.com/incident-io/catalog-importer/blob/master/client/client.go